adam/perception/perception_graph.py

```diff
@@ -2932,10 +2946,168 @@ def _translate_frames(
+                        before_label.attribute,
+                        temporal_specifiers=chain(
+                            before_label.temporal_specifiers,
+                            after_label.temporal_specifiers,
+                        ),
+                    )
+                else:
+                    _dynamic_digraph.edges[source, target][
+                        LABEL
+                    ] = TemporallyScopedEdgeLabel(
+                        after_label.attribute,
+                        temporal_specifiers=chain(after_label.temporal_specifiers),
+                    )
+                    logging.warning(
+                        f"We currently don't know how to handle a change in label "
+                        f"on an edge between the before frame and the after frame."
+                        f"Source={source}; Target={target}; "
+                        f"before label={before_label.attribute}; "
+                        f"after label={after_label.attribute}."
+                        f"As a hack, we delete the 'before' and preserve the 'after'."
+                        f"See https://github.com/isi-vista/adam/issues/666"
+                    )
+            else:
+                # This edge does not also appear in the first frame,
+                # so no merging is needed.
+                _dynamic_digraph.add_edge(source, target, label=after_label)
+
+        decoded_object_to_cluster_node = {
+            object_cluster_node.cluster_id: object_cluster_node
+            for object_cluster_node in _dynamic_digraph.nodes
+            if isinstance(object_cluster_node, ObjectClusterNode)
+        }
+
+        with open(
+            perceptual_representation.simulated_actions_file, encoding="utf-8"
+        ) as action_feature_file:
+            action_features = yaml.safe_load(action_feature_file)
+            action_stroke_graph = action_features["stroke_graph"]
+
+            # This needs to look up the nodes in the original graphs?
+            agent_node = decoded_object_to_cluster_node[action_features["object_name"]]
+            involved_object_nodes = [
+                decoded_object_to_cluster_node[object_name]
+                for object_name in action_features["objects"]
+            ]
+            involved_object_nodes.append(agent_node)
+
+            action_decode_node = CategoricalNode(
+                label="action_gnn",
+                value=action_stroke_graph["action_name"].lower(),
+                weight=action_stroke_graph["confidence_score"],
+            )
+
+            joint_point_nodes = []
+            for i, joint_feature in enumerate(
+                zip(
+                    action_stroke_graph["joint_points_world_coords"],
+                    action_stroke_graph["joint_points_xyd_coords"],
+                    action_stroke_graph["joint_points_confidence_score"],
+                )
+            ):
+                world_coord, xyd_coord, conf = joint_feature
```

Comment:
Since we don't need the joint feature itself, you can make this slightly more compact:
```suggestion
            for i, (world_coord, xyd_coord, conf) in enumerate(
                zip(
                    action_stroke_graph["joint_points_world_coords"],
                    action_stroke_graph["joint_points_xyd_coords"],
                    action_stroke_graph["joint_points_confidence_score"],
                )
            ):
```

Reply:
I need to commit this to memory for how you unpack a tuple with enumeration. Thanks!

Reply:
Welcome!